.github/workflows/scripts/prs/checkTypeLabel.js

```diff
@@ -52,66 +113,51 @@ module.exports = async ({ core, context, github }) => {
       per_page: 100,
     });
 
-    const commentFound = prComments?.find((c) =>
-      c.body.includes(
-        'Please add one type label to categorize the purpose of this PR appropriately:',
-      ),
-    );
+    const commentHandler = createCommentHandler({
+      core,
+      context,
+      github,
+      comments: prComments,
+    });
 
-    const commentLines = [];
+    // docs label is handled differently
+    if (typeLabelsFound.some((l) => l === 'docs')) {
+      core.info(`>>> 'docs' type label found`);
+
+      await commentHandler([COMMENT_FIRST_LINE.SUCCESS_COMMENT]);
+      return;
+    }
 
     if (typeLabelsFound.length === 0) {
       core.info(`>>> No type labels found`);
 
-      if (commentFound) {
-        core.info(`>>> PR already has the type label comment.`);
-        core.info(`>>> Exiting gracefully! 👍`);
-        return;
-      }
-
       // Add a comment line explaining that a type label needs to be added
-      commentLines.push(
-        'Please add one type label to categorize the purpose of this PR appropriately:',
-      );
-      commentLines.push(createEnumerationFromArray(typeLabels));
-    } else if (typeLabelsFound.length > 1) {
+      await commentHandler([
+        COMMENT_FIRST_LINE.NO_LABELS_COMMENT,
+        createEnumerationFromArray(typeLabels),
+      ]);
+
+      core.setFailed('>>> Failing workflow to prevent merge without passing this!');
+      return;
+    }
+
+    if (typeLabelsFound.length > 1) {
       core.info(`>>> Multiple type labels found: ${typeLabelsFound.join(', ')}`);
 
       // add a comment line explaining that only one type label is allowed
-      commentLines.push(`Multiple type labels found: ${typeLabelsFound.join(', ')}`);
-      commentLines.push(
+      await commentHandler([
+        COMMENT_FIRST_LINE.MULTIPLE_LABELS_COMMENT,
+        typeLabelsFound.map((label) => `- ${label}`).join('\n'),
```

Comment:
Changed the message so we can properly check for `commentFound.body?.startsWith(commentLines[0])` above